arm9/src/unk_0200BB14.c

```diff
@@ -0,0 +1,1280 @@
+
+THUMB_FUNC u32 FUN_0200C600(u32 param0)
+{
+    return FUN_02020388(param0);
+}
+
+THUMB_FUNC u32 FUN_0200C608(u32 *param0)
+{
+    return FUN_0200C600(*param0);
+}
+
+THUMB_FUNC void FUN_0200C614(u32 param0)
+{
+    FUN_02020208(param0);
+}
+
+THUMB_FUNC void FUN_0200C61C(u32 *param0)
+{
+    FUN_0200C614(*param0);
+}
+
+THUMB_FUNC void FUN_0200C628(u32 param0)
+{
+    FUN_0202022C(param0);
+}
+
+THUMB_FUNC void FUN_0200C630(u32 *param0)
+{
+    FUN_0200C628(*param0);
+}
+
+THUMB_FUNC void FUN_0200C63C(u32 param0, u32 param1)
+{
+    FUN_020200A0(param0, param1);
+}
+
+THUMB_FUNC void FUN_0200C644(u32 *param0, u32 param1)
+{
+    FUN_0200C63C(*param0, param1);
+}
+
+THUMB_FUNC u8 FUN_0200C650(u32 param0)
+{
+    return FUN_02020128(param0);
+}
+
+THUMB_FUNC u8 FUN_0200C658(u32 *param0)
+{
+    return FUN_0200C650(*param0);
+}
+
+THUMB_FUNC void FUN_0200C664(u32 param0, u32 param1)
+{
+    FUN_02020248(param0, param1);
+}
+
+THUMB_FUNC void FUN_0200C66C(u32 *param0, u32 param1)
+{
+    FUN_0200C664(*param0, param1);
+}
+
+THUMB_FUNC void FUN_0200C678(u32 param0, u32 param1)
+{
+    FUN_020202A8(param0, param1);
+}
+
+THUMB_FUNC void FUN_0200C680(u32 *param0, u32 param1)
+{
+    FUN_0200C678(*param0, param1);
+}
+
+THUMB_FUNC u8 FUN_0200C68C(u32 *param0)
+{
+    return FUN_02020300(*param0);
+}
+
+THUMB_FUNC void FUN_0200C698(u32 param0, u32 param1)
+{
+    FUN_02020238(param0, (u8)param1);
+}
+
+THUMB_FUNC u8 FUN_0200C6A4(u32 *param0)
+{
+    return FUN_02020240(*param0);
+}
+
+THUMB_FUNC void FUN_0200C6B0(u32 *param0, u32 param1)
+{
+    FUN_0200C698(*param0, param1);
+}
+
+THUMB_FUNC void FUN_0200C6BC(u32 param0, u32 param1)
+{
+    FUN_02020310(param0, param1);
+}
+
+THUMB_FUNC void FUN_0200C6C4(u32 *param0, u32 param1)
+{
+    FUN_0200C6BC(*param0, param1);
+}
+
+THUMB_FUNC u16 FUN_0200C6D0(u32 param0)
+{
+    return FUN_0202032C(param0);
+}
+
+THUMB_FUNC u16 FUN_0200C6D8(u32 *param0)
+{
+    return FUN_0200C6D0(*param0);
+}
+
+THUMB_FUNC void FUN_0200C6E4(u32 param0, u32 param1, u32 param2)
+{
+    u32 st0[3];
+    st0[0] = param1 << 12;
+    st0[1] = param2 << 12;
+    if (FUN_02020380(param0) == 2)
+    {
+        st0[1] += 0xc0000;
+    }
+    st0[2] = 0;
+
+    FUN_02020044(param0, st0);
+}
+
+THUMB_FUNC void FUN_0200C714(u32 *param0, u32 param1, u32 param2)
+{
+    FUN_0200C6E4(*param0, param1, param2);
+}
+
+THUMB_FUNC void FUN_0200C720(u32 param0, u32 param1, u32 param2, u32 param3)
+{
+    u32 st0[3];
+
+    st0[0] = param1 << 12;
+    st0[1] = param2 << 12;
+    if (FUN_02020380(param0) == 2)
+    {
+        st0[1] += param3;
+    }
+    st0[2] = 0;
+
+    FUN_02020044(param0, st0);
+}
+
+THUMB_FUNC void FUN_0200C750(u32 *param0, u32 param1, u32 param2, u32 param3)
+{
+    FUN_0200C720(*param0, param1, param2, param3);
+}
+
+THUMB_FUNC void FUN_0200C75C(u32 param0, u16 *param1, u16 *param2)
+{
+    u32 *r4 = FUN_0202011C(param0, param1, param2);
+
+    param1[0] = (u16)((int)(r4[0] + ((u32)((int)r4[0] >> 0xb) >> 0x14)) >> 0xc);
```

Comment:
I feel like half of these casts are unnecessary

Reply:
This is actually a signed division

Reply:
then why don't you put it as a signed division instead of well, all this

Reply:
Ahah, I just found out signed divisions were implemented with all this crazy shifts by trying to clean it up. I didn't know that before.